### PR TITLE
22w15a more stuff

### PR DESCRIPTION
**Pull request by apple502j**

### CODE REVIEW

File: mappings/net/minecraft/datafixer/fix/BlendingDataFix.mapping [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+	FIELD field_37415 SKIP_BLENDING_STATUSES Ljava/util/Set;
 	METHOD <init> (Lcom/mojang/datafixers/schemas/Schema;)V
 		ARG 1 schema
+	METHOD method_41311 (Lcom/mojang/datafixers/Typed;)Lcom/mojang/datafixers/Typed;
+		ARG 0 typed
+	METHOD method_41312 update (Lcom/mojang/serialization/Dynamic;)Lcom/mojang/serialization/Dynamic;
+		ARG 0 dynamic
+	METHOD method_41313 setSections (Lcom/mojang/serialization/Dynamic;II)Lcom/mojang/serialization/Dynamic;
+		ARG 0 dynamic
+		ARG 1 maxY
```
Comment (enbrain) [inline mappings/net/minecraft/datafixer/fix/BlendingDataFix.mapping]:
```suggestion
		ARG 1 height
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 2]
```diff
@@ -226,7 +226,15 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 	METHOD method_23311 refreshPosition ()V
-	METHOD method_23312 getLandingPos ()Lnet/minecraft/class_2338;
+	METHOD method_23312 getSteppingPos ()Lnet/minecraft/class_2338;
+		COMMENT {@return the stepping position}
+		COMMENT
+		COMMENT @implNote Stepping position is the entity's position, with {@code 1e-05} subtracted
+		COMMENT from the Y coordinate. This means that, for example, if a player is on a carpet on
+		COMMENT a soul soil, the carpet's position would be returned.
+		COMMENT
+		COMMENT @see getLandingPos()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getLandingPos()
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 3]
```diff
@@ -226,7 +226,15 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see getSteppingBlockState()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getSteppingBlockState()
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 4]
```diff
@@ -270,7 +278,14 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 		ARG 1 onGround
-	METHOD method_25936 getLandingBlockState ()Lnet/minecraft/class_2680;
+	METHOD method_25936 getSteppingBlockState ()Lnet/minecraft/class_2680;
+		COMMENT {@return the block state at the stepping position}
+		COMMENT
+		COMMENT @implNote Stepping position is the entity's position, with {@code 1e-05} subtracted
+		COMMENT from the Y coordinate. This means that, for example, if a player is on a carpet on
+		COMMENT a soul soil, the carpet's position would be returned.
+		COMMENT
+		COMMENT @see getSteppingPos()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getSteppingPos()
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 5]
```diff
@@ -446,9 +461,28 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
 		ARG 2 horizontalRadius
 		ARG 4 verticalRadius
+	METHOD method_43260 getLandingPos ()Lnet/minecraft/class_2338;
+		COMMENT {@return the landing position}
+		COMMENT
+		COMMENT @implNote Landing position is the entity's position, with {@code 0.2} subtracted
+		COMMENT from the Y coordinate. This means that, for example, if a player is on a carpet on
+		COMMENT a soul soil, the soul soil's position would be returned.
+		COMMENT
+		COMMENT @see getSteppingPos()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getSteppingPos()
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 6]
```diff
@@ -446,9 +461,28 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+		COMMENT @see getLandingBlockState()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getLandingBlockState()
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 7]
```diff
@@ -446,9 +461,28 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
...
+	METHOD method_43261 getLandingBlockState ()Lnet/minecraft/class_2680;
+		COMMENT {@return the block state at the landing position}
+		COMMENT
+		COMMENT @implNote Landing position is the entity's position, with {@code 0.2} subtracted
+		COMMENT from the Y coordinate. This means that, for example, if a player is on a carpet on
+		COMMENT a soul soil, the soul soil's position would be returned.
+		COMMENT
+		COMMENT @see getLandingPos()
```
Comment (enbrain) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT @see #getLandingPos()
```

File: mappings/net/minecraft/data/DataGenerator.mapping [hunk 8]
```diff
@@ -3,7 +3,17 @@ CLASS net/minecraft/class_2403 net/minecraft/data/DataGenerator
 	FIELD field_11273 providers Ljava/util/List;
 	FIELD field_11274 output Ljava/nio/file/Path;
 	FIELD field_11275 LOGGER Lorg/slf4j/Logger;
+	FIELD field_38909 runningProviders Ljava/util/List;
+	FIELD field_38910 gameVersion Lnet/minecraft/class_6489;
+	FIELD field_38911 skipGeneratorRerunWarning Z
```
Comment (enbrain) [inline mappings/net/minecraft/data/DataGenerator.mapping]: This is more than skipping warning. If set to `true`, it runs data providers even if they were already run. Maybe `ignoreCache`?

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/data/DataGenerator.mapping (hunk 8 above)
Comment (apple502j) [reply mappings/net/minecraft/data/DataGenerator.mapping]:
```suggestion
	FIELD field_38911 ignoreCache Z
```

File: mappings/net/minecraft/data/DataGenerator.mapping [hunk 9]
```diff
@@ -3,7 +3,17 @@ CLASS net/minecraft/class_2403 net/minecraft/data/DataGenerator
...
 	FIELD field_11274 output Ljava/nio/file/Path;
 	FIELD field_11275 LOGGER Lorg/slf4j/Logger;
+	FIELD field_38909 runningProviders Ljava/util/List;
+	FIELD field_38910 gameVersion Lnet/minecraft/class_6489;
+	FIELD field_38911 ignoreCache Z
+	METHOD <init> (Ljava/nio/file/Path;Ljava/util/Collection;Lnet/minecraft/class_6489;Z)V
+		ARG 1 output
+		ARG 2 inputs
+		ARG 3 gameVersion
+		ARG 4 skipGeneratorRerunWarning
```
Comment (enbrain) [inline mappings/net/minecraft/data/DataGenerator.mapping]:
```suggestion
		ARG 4 ignoreCache
```

File: mappings/net/minecraft/data/Main.mapping [hunk 10]
```diff
@@ -9,3 +11,5 @@ CLASS net/minecraft/data/Main
 		ARG 4 includeDev
 		ARG 5 includeReports
 		ARG 6 validate
+		ARG 7 gameVersion
+		ARG 8 skipGeneratorRerunWarning
```
Comment (enbrain) [inline mappings/net/minecraft/data/Main.mapping]:
```suggestion
		ARG 8 ignoreCache
```

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**